### PR TITLE
feat: Add match_action for plugin configs

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Just the migration part of https://github.com/PostHog/posthog/pull/22056

## Changes

* Adds new match action field for plugins

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: posthog/management/commands/test_migrations_are_safe.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def validate_migration_sql(sql) -> bool:
             )  # Ignore for brand-new tables
         ):
             print(
-                f"\n\n\033[91mFound a CONSTRAINT command. This locks tables which causes downtime. Please avoid adding constraints to existing tables.\nSource: `{operation_sql}`"
+                f"\n\n\033[91mFound a CONSTRAINT command. This locks tables which causes downtime. Please avoid adding constraints to existing tables. For an example of how to do this safely see 0412_pluginconfig_match_action.py . \nSource: `{operation_sql}`"
```
Comment (neilkakkar) [inline posthog/management/commands/test_migrations_are_safe.py:76]: I think you want this on line 85 as well - where we have the create index concurrently check.

The NULL constraint I think is just boilerplate, I'd say we can go as far as removing it from this check, and in that case the error would be on the create index command.

Comment (neilkakkar) [reply posthog/management/commands/test_migrations_are_safe.py:76]: image

**Review by neilkakkar — APPROVED** [review]
Nice looks good! Just one commend to add

File: posthog/management/commands/test_migrations_are_safe.py (hunk 1 above)
Comment (benjackwhite) [reply posthog/management/commands/test_migrations_are_safe.py]: Added it in the other place too

**Pull request merged**